### PR TITLE
deleted selection 'no work needed', all recordings are available now …

### PR DESCRIPTION
**Pull request by ElsDeBie**
on demand of Patrik O to adjust this function so also relevés with work can be selected. The user can make an extra selection by only chosing need work = 0 to only select finished relevés.

### CODE REVIEW

File: R/get_inboveg_header.R
Comment (hansvancalster) [inline R/get_inboveg_header.R:118]: removing the `where` statement will give trouble because other `where` statements are pasted to the `common_part` of the sql statement. Maybe try changing old line 115 to `where 1=1`

Comment (ElsLommelen) [reply R/get_inboveg_header.R:118]: Another option would be to replace the `AND` at the beginning of the 2 pasted parts by `WHERE`.  Advantage is that we wouldn't ask ourselves in the future why this `1=1` is added (and it is less verbose).

Comment (ElsDeBie) [reply R/get_inboveg_header.R:118]: is there a tric to control the functions you adapted? now I always copy them in a seperate r-file and try to run, but often this need a lot of adjustments (it works of course, but more efficient would be when testing is possible in this project itself).

Comment (ElsDeBie) [reply R/get_inboveg_header.R:118]: @ElsLommelen "where 1 = 1 " is often used in sql-scripts (allez dat is wat Jo Loos me eigenlijk zo geleerd heeft, dus dom dat ik daar hier niet zelf aan gedacht had).

**Comment by ElsLommelen** [comment]
@ElsDeBie When finished, can you please let me know, so I can add the documentation to the website?  Or I can just let you know how to do it, if you wish, up to you to choose...

File: R/get_inboveg_header.R
Comment (ElsLommelen) [reply R/get_inboveg_header.R:118]: Normally you have in this project in your right upper corner a tab 'build'.  Clicking the button 'install and restart' installs the adapted version of the package on your computer.  And then you can for instance run the example (in the documentation above your function) by putting your cursor on it and hit crtl-enter.

Comment (ElsLommelen) [reply R/get_inboveg_header.R:118]: Ah, ok, wist ik niet, en de reden is gewoon om die volgende regel niet te moeten aanpassen?

**Comment by ElsDeBie** [comment]
this part is finished, I guess. so can be update. Or which documentation do you mean? the vignettes?

**Comment by ElsLommelen** [comment]
Both the documentation which is shown in R after you hit `?get_inboveg_header` and the documentation on the website.  So after you adapt it above the function itself, these 2 derived outputs should be generated.  (I plan to automate this in near future, but as long as it is not done, it should be done 'manually' by hitting a button and writing a simple command.)

**Comment by ElsLommelen** [comment]
Ok, done, I suppose it can be merged?

**Comment by ElsDeBie** [comment]
Can you give me a hint? was it with knitr? I wrote it down somewhere. Is it somewhere  in a manual?

**Comment by ElsLommelen** [comment]
To update the documentation in R, I usually hit the 'check' button on the build table, as it is a good habit to do these checks as well, and the documentation is build in the same time (except for the vignet, but you haven't changed that now).  But strictly, to only update this documentation (including vignettes), one should use the command `devtools::install(build_vignettes = TRUE)`.

Update the website is done with `pkgdown::build_site()`.

I've no idea if it is somewhere in a manual, the easiest is to just setup the repo to do it automatically (which has at least the advantage that this update is not forgotten).  Thierry is now developing the package [checklist](https://packages.inbo.be/checklist/) that does this setup for you (together with some checks if the package works properly, website in INBO style,...).  So that's from now on the easiest way for starting a new package with all necessary things included, I guess.  Unfortunately it did not yet exist when we started inbodb, and setup it afterwards requires some time to adjust the package to the requirements in the tests, or trying to ignore some of the tests (which is in fact more complicated than just adjusting the package).  But as soon as I have some time to finish this at once, I'll tackle this.

**Comment by hansvancalster** [comment]
> To update the documentation in R, I usually hit the 'check' button on the build table, as it is a good habit to do these checks as well, and the documentation is build in the same time (except for the vignet, but you haven't changed that now). But strictly, to only update this documentation (including vignettes), one should use the command `devtools::install(build_vignettes = TRUE)`.

Or `devtools::document()` to only update documentation of functions and namespace (without vignettes)

**Comment by ElsLommelen** [comment]
I guess everyone has read the discussion, so I can merge the branch?  ;-)

**Review by hansvancalster — APPROVED** [review]
This looks OK to me and is ready to be merged. Just one suggestion: add an explanation in the [vignette](https://github.com/inbo/inbodb/blob/master/vignettes/get_data_inboveg.Rmd) about the meaning of the `needsWork` column. I leave it up to @ElsDeBie to make this change in this PR or first merge this PR and start a separate PR in which the vignette is updated.

**Comment by ElsDeBie** [comment]
I will first solve the remark of Hans (good one!).

**Comment by ElsDeBie** [comment]
How can I adjust the table in the vignette? the example of head(header_info, 10), don't show the last column 'work needed'. But the function works as if you can see below the function (after running that chunk). Or do I have to adjust it manually? 15columns  instead of 16 in running chunk code.

image

File: vignettes/get_data_inboveg.Rmd
Comment (ElsLommelen) [inline vignettes/get_data_inboveg.Rmd]:
```suggestion
More information? Check out [https://www.vlaanderen.be/inbo/en-gb/data-applications/inboveg/](https://www.vlaanderen.be/inbo/en-gb/data-applications/inboveg/)
```

Like this you can make it a link.  If you wish, you change the text on the link by changing the text inside the `[  ]`.

**Comment by hansvancalster** [comment]
> How can I adjust the table in the vignette? the example of head(header_info, 10), don't show the last column 'work needed'. But the function works as if you can see below the function (after running that chunk). Or do I have to adjust it manually? 15columns instead of 16 in running chunk code.
> 


The chunk options are set to `eval = FALSE` in that vignette. So the code will not be evaluated when the vignette is built. Therefore, the idea is to copy-paste the output into the chunk as commented out text beneath the code.
@ElsLommelen this is a workaround to avoid that the vignette when built has no connection to the databases. But it's awkward. Can't we think of a safe (i.e. no errors or warnings during `devtools::check()`) way to built the vignette _and_ evaluate the code during build?

**Comment by ElsLommelen** [comment]
> Can't we think of a safe (i.e. no errors or warnings during `devtools::check()`) way to built the vignette _and_ evaluate the code during build?

As far as I know, the only way to do this (kept in mind that I want to automate the whole build and check process with github actions), is to add a small example database in the package, or code to generate a test database.

**Comment by hansvancalster** [comment]
> > Can't we think of a safe (i.e. no errors or warnings during `devtools::check()`) way to built the vignette _and_ evaluate the code during build?
> 
> As far as I know, the only way to do this (kept in mind that I want to automate the whole build and check process with github actions), is to add a small example database in the package, or code to generate a test database.

[this answer](https://community.rstudio.com/t/precompiling-vignette-with-devtools/1583/6) looks like a solution that could work. Would you mind trying it?

**Comment by ElsLommelen** [comment]
Euh, I'm not sure I understand you, are you sure you added the correct link?

**Comment by hansvancalster** [comment]
> Euh, I'm not sure I understand you, are you sure you added the correct link?

😂 that was the wrong link indeed. I've edited the post, so it should now link to the correct website.

**Comment by ElsLommelen** [comment]
Ok, I will have a look at it, but not now.  I made an issue for the automation and then I will tackle this as well.  Feel free to adapt the code already, for now there are no online checks in this package  (And don't forget to rebuild the website before merging.)

**Comment by hansvancalster** [comment]
@ElsDeBie That means you can remove these `#> comments` from all chunks in the  vignette. 
You just have to make sure that what you see as the output from a chunk is what is most instructive for the readers. So if you talk about `needsWork` it might be a good idea to have a chunk that shows a count of the different values in that column.

**Comment by ElsDeBie** [comment]
there seems to be a problem with a package called xfun ... I will remove it and than try to install it again via R instead of Rstudio.

**Comment by ElsDeBie** [comment]
Got this message when i did the 'check' in 'build:

Error in get(genname, envir = envir) : object 'testthat_print' not found
Updating inbodb documentation
... then everything went well untill the end:
-- R CMD check results --------------------------------------- inbodb 0.0.1 ----
Duration: 1m 2.1s

> checking for unstated dependencies in examples ... OK
   WARNING
  'qpdf' is needed for checks on size reduction of PDFs

0 errors v | 1 warning x | 0 notes v
Error: R CMD check found WARNINGs
Execution halted

Exited with status 1.
#######################
After running : > devtools::install(build_vignettes = TRUE)
Error in get(genname, envir = envir) : object 'testthat_print' not found
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

 1: All 
...
package ‘tibble’ successfully unpacked and MD5 sums checked
Error: (converted from warning) cannot remove prior installation of package ‘tibble’

This afternoon I had a problem with xfun, which I fixed by remove.packages and install.packages.
Is it normal I always have to update 'all' packages? It seems like they are not really update ?

**Comment by ElsLommelen** [comment]
@ElsDeBie  Do you have package `testthat` installed?

For updating packages: you have to do it after an update of R on you computer, but he always suggests to do it whenever there are newer versions available.  I always try my packages to be backwards compatible up to the versions that were available at the previous update of R at INBO, and that's why I avoid these updates if not strictly necessary (but I actually should better tackle this by automatic checks with previous versions).  Recently I once encountered a problem due to these partial updates: I had updated some packages because I had to (or because I installed a new package), and because other packages were not updated, they did not perform well together anymore and RStudio regularly crashed.  I finally solved this by updating all packages.  So in case you regularly update some packages when asked, it is strongly advised to update all packages together once in a while (following the tutorial instructions on what to do after an update of R).

**Comment by ElsDeBie** [comment]
@ElsLommelen  thanks for the comprehensive answer. my pc was updated to windows 10. could this cause the crashes of Rstudio every now and then? 
And yes in my library 'testthat' is installed. or at least it was in c:/R/library, but when I wanted to open it, I got: 

> library(testthat)
Error in get(genname, envir = envir) : object 'testthat_print' not found

Restarting R session...

> install.packages("testthat")
...
The downloaded binary packages are in
	C:\Users\els_debie\AppData\Local\Temp\RtmpQBK190\downloaded_packages
But when I install package this is the window that pops up:
image
so the paht to my library is c:/R/library
Can this cause the conflict? Or should I search somewhere else (global options seems okay)

**Comment by ElsLommelen** [comment]
Apparently a problem that you combine an old version of `testthat` and a new version of `rlang` (explained [here](https://github.com/r-lib/devtools/issues/2309)), and as I explained, also the crashes could have to do with the fact that the installed versions of your packages are not compatible.  Use [this manual](https://inbo.github.io/tutorials/installation/user/user_install_r/) to update all packages to the newest version (which makes sure they are compatible again) and make sure your RStudio refers to the latest installed R version (see [here below](https://inbo.github.io/tutorials/installation/user/user_install_rstudio/)).

I'm not on Windows 10 yet, so I don't know if this could have anything to do with it.  Normally I would expect that you have recently installed/updated all packages, by which you should have less problems.

Normally everyone at INBO uses C:/R/library for the libraries, so this should be ok.

File: vignettes/get_data_inboveg.Rmd
Comment (hansvancalster) [inline vignettes/get_data_inboveg.Rmd]: @ElsDeBie I would just remove all such lines (the commented out output) from the r-chunks (not only here, in all chunks). @ElsLommelen will work out a way that the vignette can be build locally and does not give problems with package checks.

Comment (ElsLommelen) [reply vignettes/get_data_inboveg.Rmd]: And if you wish to show the output, you can already remove the `eval = FALSE` in the first chunk.  This means we have to build the website locally when the vignette has changed, but this will stay the case with the option suggested by @hansvancalster.

Comment (ElsDeBie) [reply vignettes/get_data_inboveg.Rmd]: I now notice that sometimes there is a code to show the result, and sometimes there is just a code without showing the result. What is the best to do? for example
```{r examples_survey}
survey_info <- get_inboveg_survey(con, 
                              survey_name = "OudeLanden_1979", 
                              collect = TRUE)
```
```{r}
survey_info
```
so the result is shown, (isn't it beter to put both in one chunck?)

or 
```{r}
header_severalsurveys <- get_inboveg_header(con,
                survey_name = c("MILKLIM_Heischraal2012", "NICHE Vlaanderen"),
                multiple = TRUE)
```
and then a next example is shown without given the result of "header_severalsurveys"

Comment (ElsLommelen) [reply vignettes/get_data_inboveg.Rmd]: It depends on the purpose and what you are explaining/illustrating, I think.  If the focus in the text is on the output, it is definitively necessary to show the output.  If you just want to illustrate some different coding possibilities (all of them having more or less the same output), it may not always be necessary to show the full output for each of them.

So I would check in the text what seems appropriate to you, and keep in mind that users may read this manual before deciding on using the package.  (Different from the examples at the help functions, this is a document to read, so users are probably less likely to try the code, and expect the results to be shown whenever appropriate.)

**Comment by ElsDeBie** [comment]
can I continue working in branch expands functions, or should I wait when this branch header is merged into the master?

**Comment by ElsLommelen** [comment]
Theoretically you can, but if it's a good idea, mainly depends on whether or not you make changes on the same lines you did here, or lines directly above or below.  If you are working on different files, or completely different parts of a file, then you can do so without having any problems afterwards.

Are you finished here?  Then I'll update the website documentation before we forget to do so.  ;-)

**Comment by ElsDeBie** [comment]
yes, I 'm finished. 
And can this also be merged then into the master ... so I can work an my other branch to create new functions.

**Comment by ElsLommelen** [comment]
I could merge, but I notice that running the code chunks in the vignette has a similar appearance as the 'commented code' where @hansvancalster referred to.  I don't know what to do with it, merge so you can continue, or wait for @hansvancalster for his opinion?

On the other hand, @hansvancalster approved the PR.  As this commented code was not the main subject of the PR, I suggest @hansvancalster to start a PR on it if he wants to have it changed.  This because it is also unclear to me if he just wanted to get rid of the pasted code or if he didn't like the appearance of it.

**Pull request merged**
